### PR TITLE
Specified AWS and Azure params for the Terraform provider

### PR DESCRIPTION
**Pull request by AdminTurnedDevOps**
Addition of AWS access key and secret key fields for the Terraform provider.

### CODE REVIEW

**Review by droyad — CHANGES_REQUESTED** [review]
AWS accounts should be it's own resource type like `octopusdeploy_azure_service_principal`

File: octopusdeploy/resource_account.go [hunk 1]
```diff
@@ -26,6 +26,7 @@ func resourceAccount() *schema.Resource {
 				},
 				Optional: true,
 			},
+			// Azure
```
Comment (droyad) [inline octopusdeploy/resource_account.go:29]: `account_type` applies to all accounts, it's the thing that determines whether it's Azure/AWS/UsernamePasword, etc

File: octopusdeploy/resource_account.go [hunk 2]
```diff
@@ -66,6 +67,15 @@ func resourceAccount() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 			},
+			// AWS
```
Comment (droyad) [inline octopusdeploy/resource_account.go:70]: Are the changes to this file still required? Should we remove `resource_account.go`?

File: octopusdeploy/resource_usernamepassword_account.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"environments": {
+				Type: schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional: true,
+			},
```
Comment (droyad) [inline octopusdeploy/resource_usernamepassword_account.go:32]: The tenant scoping is missing from this resource. I think the common elements of Account should be moved into re-used method like we did with the [actions](https://github.com/OctopusDeploy/terraform-provider-octopusdeploy/blob/0ecd0b5184f2b850c9a507c7d340e1b7ec6fbb17/octopusdeploy/manual_intervention_action.go#L9).

**Review by droyad — CHANGES_REQUESTED** [review]

File: octopusdeploy/resource_account.go (hunk 1 above)
Comment (AdminTurnedDevOps) [reply octopusdeploy/resource_account.go]: I removed those as they are no longer needed.

File: octopusdeploy/resource_account.go (hunk 2 above)
Comment (AdminTurnedDevOps) [reply octopusdeploy/resource_account.go]: I removed those as they are no longer needed @droyad

File: octopusdeploy/resource_usernamepassword_account.go (hunk 3 above)
Comment (AdminTurnedDevOps) [reply octopusdeploy/resource_usernamepassword_account.go:37]: Sounds good. I added back in the tenant so it'll use `getTenantedDeploymentSchema()`

**Comment by AdminTurnedDevOps** [comment]
@droyad Are these changes all good for you?

**Pull request merged**